jupyter_server/utils.py

```diff
@@ -398,3 +400,47 @@ def maybe_future(obj):
         f.set_result(obj)
         return f
 
+
+def authorized(action, resource=None, message=None):
+    """A decorator for tornado.web.RequestHandler methods
+    that verifies whether the current user is authorized
+    to make the following request.
+
+    Helpful for adding an 'authorization' layer to
+    a REST API.
+
+    Parameters
+    ----------
+    action : str
+        the type of permission or action to check.
+
+    resource: str or None
+        the name of the resource the action is being authorized
+        to access.
+
+    message : str or none
+        a message for the unauthorized action.
+    """
+    # Get message
+    if message is None:
+        "User is not authorized to make this request."
```

Comment:
Should this be assigned to a message variable?
```
message = "User is not authorized to make this request."
```
Also I think it would be helpful for debugging to know what particular action a user is not authorized to do.
```
f"User is not authorized to perform action: {action} on resource type: {resource}."
```